tests-integration/src/test_util.rs

```diff
@@ -13,44 +13,152 @@
 // limitations under the License.
 
 use std::collections::HashMap;
+use std::env;
+use std::net::SocketAddr;
+use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
+use std::time::Duration;
 
+use axum::Router;
 use catalog::CatalogManagerRef;
 use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME, MIN_USER_TABLE_ID};
+use common_runtime::Builder as RuntimeBuilder;
 use datanode::datanode::{DatanodeOptions, ObjectStoreConfig};
 use datanode::error::{CreateTableSnafu, Result};
+use datanode::instance::{Instance, InstanceRef};
 use datanode::sql::SqlHandler;
 use datatypes::data_type::ConcreteDataType;
 use datatypes::schema::{ColumnSchema, SchemaBuilder};
+use frontend::frontend::FrontendOptions;
+use frontend::grpc::GrpcOptions;
+use frontend::instance::{FrontendInstance, Instance as FeInstance};
+use object_store::backend::s3;
+use object_store::test_util::TempFolder;
+use object_store::ObjectStore;
+use once_cell::sync::OnceCell;
+use rand::Rng;
+use servers::grpc::GrpcServer;
+use servers::http::{HttpOptions, HttpServer};
+use servers::server::Server;
 use servers::Mode;
 use snafu::ResultExt;
 use table::engine::{EngineContext, TableEngineRef};
 use table::requests::CreateTableRequest;
 use tempdir::TempDir;
+static PORTS: OnceCell<AtomicUsize> = OnceCell::new();
+
+fn get_port() -> usize {
```

Comment:
Seems the port is used to start the testing grpc server, maybe we can use port 0?

Reply:
Some ports need system permissions,  so we have to limit them to a range.